source/MRViewer/MROpenObjects.cpp

```diff
@@ -215,25 +221,41 @@ Expected<LoadedObject> makeObjectTreeFromFolder( const std::filesystem::path & f
         }
         else
         {
-            ++allErrors[taskRes.error()];
+            if ( auto it = allErrors.find( taskRes.error() ); it != allErrors.end() )
+            {
+                it->second.count += 1;
+            }
+            else
+            {
+                std::error_code ec;
+                allErrors[taskRes.error()] = { 1, utf8string( std::filesystem::relative( node.path, folder, ec ) ) };
+                if ( ec )
+                    spdlog::warn( "Filesystem error when trying to obtain {} relative to {}: {}", utf8string( node.path ), utf8string( folder ), ec.message() );
+            }
         }
     }
 
     std::string errorString;
     for ( const auto& error : allErrors )
     {
         errorString += ( errorString.empty() ? "" : "\n" ) + error.first;
-        if ( error.second > 1 )
+        if ( error.second.count > 1 )
         {
-            errorString += std::string( " (" ) + std::to_string( error.second ) + std::string( ")" );
+            errorString += std::string( " (" ) + std::to_string( error.second.count ) + std::string( ")" );
         }
+        else
+            errorString += std::string( " (for " ) + utf8string( error.second.path ) + std::string( ")" );
     }
 
-    if ( !errorString.empty() )
-        spdlog::warn( "Load folder error:\n{}", errorString );
     if ( !atLeastOneLoaded )
         return unexpected( errorString );
 
+    if ( !errorString.empty() )
+    {
+        spdlog::warn( "Load folder error:\n{}", errorString );
+        res.warnings = errorString + '\n' + res.warnings;
```

Comment:
Why do you add at front and not at back?

Reply:
Because errors are more important than warnings.